pkg/services/provisioning/channelrule/provision.go

```diff
@@ -0,0 +1,110 @@
+package channelrule
+
+import (
+	"context"
+	"errors"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/live/database"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+// Provision scans a directory for provisioning config files
+// and provisions the channelRule in those files.
+func Provision(configDirectory string, sqlStore *sqlstore.SQLStore) error {
```

Comment:
Not optimal to extend the existing provisioning solution, but there's no other alternative currently why we accept it. I would recommend splitting this PR in two, keeping the provisioning separate to make it easier to review and getting things merged. Provisioning documentation is lacking, but maybe you're intending to add that later?

In addition, you need to add provisioning folder to the packaging process, see #34398 for an example.

Reply:
Thanks, fixed

To be fair I am not sure I'll be able to split this into 2 separate pull requests since commits are not ordered here, is there any technique to do this on folder basis?

Reply:
Will submit docs later in another pr

Reply:
I guess you have to manually remove the provisioning changes from this branch and push changes to a new branch.